pandas/core/indexing.py

```diff
@@ -165,7 +166,37 @@ def _get_loc(self, key: int, axis: int):
     def _slice(self, obj, axis: int, kind=None):
         return self.obj._slice(obj, axis=axis, kind=kind)
 
+    def _ensure_listlike_indexer(self, key, axis: int):
```

Comment:
type key (ok for a followon)

Reply:
Not sure how to type key as it can be any listlike type. I was just following the type of `_get_listlike_indexer` which leaves key untyped too.

Reply:
well, it would be really great to type the indexing keys

```Indexable=Iterable``` might be enough for now (you can add/import from pandas._typing)

can this be a scalar?

Reply:
> can this be a scalar?

No.